src/experimental/job/job.ts

```diff
@@ -151,7 +151,7 @@ export class Job<Output = unknown> {
 
     signal.addEventListener("abort", onAbort, { once: true });
 
-    return workOnGolem(exeUnit);
+    return workOnGolem(exeUnit).finally(() => rental.stopAndFinalize());
```

Comment:
IMO this should be moved to the `finally` block on 127 and we should only emit the `ended` event once the rental is actually finalized

Reply:
there is no information about the `rental` in the `finally` block (#127), it has not been created yet. I added `try / finally` block and `await` before `finalize` so that event`ended` would be emitted always after rental is finalized

Reply:
ah you're right, my mistake. I like the change you did. I think we don't event need to call `waitAndFinalize` in `onAbort` because it will be called anyways after the error is thrown